app/forms/hub/client_form.rb

```diff
@@ -0,0 +1,46 @@
+module Hub
+  class ClientForm < Form
+    include FormAttributes
+    set_attributes_for :intake,
+        :sms_phone_number,
+        :phone_number,
+        :preferred_name,
+        :primary_first_name,
+        :primary_last_name,
+        :email_address,
+        :preferred_interview_language,
+        :sms_notification_opt_in,
+        :email_notification_opt_in
+
+    before_validation do
+      self.sms_phone_number = PhoneParser.normalize(sms_phone_number)
+      self.phone_number = PhoneParser.normalize(phone_number)
+      self.preferred_name = preferred_name.presence || "#{primary_first_name} #{primary_last_name}"
```

Comment:
When you go through the client-side flow, the first questions we ask are for "name" and "zip code" -- even though we don't specify, I think the UI is intended to imply that you enter your full name there since you're starting an official form to get your taxes done.

Because we're using the preferred name field as the display name on the clients index _and_ its currently the only clickable field to the show page, it seems like we are expecting it to be a first and last name -- it may be the case that in fact we want to be using Legal name on the client index and we should update that.

Alex and I actually did the update to preferred "full" name because we felt like it was _less_ clear on the manual create client form that the preferred name should be the users entire preferred name (first and last, or something more than a nickname).


I'm going to share this conversation with product so that we can figure out how we want to treat names moving forward.